### PR TITLE
Add docker improvements

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Moving environment variables and ports in the docker compose override file, creating a make command, some tweaks on README
| Type?             | improvement
| Category?         | IN
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | -
| Fixed ticket?     | -
| Related PRs       | -
| Sponsor company   | -

### CODE REVIEW

**Review by kpodemski — DISMISSED** [review]

File: docker-compose.yml
Comment (boherm) [inline docker-compose.yml]: Hum, I'm not sure why you have to switch this port. Can you explain?

Comment (M0rgan01) [reply docker-compose.yml]: Of course :smile: 

If you specify only the container port, an ephemeral port will be created for the host. 
This is currently the case.

I propose to specify the port of the host, here 3307. This avoids going to look for the port chosen by docker.

See https://docs.docker.com/compose/compose-file/compose-file-v3/#ports for more informations

**Review by FabienPapet — CHANGES_REQUESTED** [review]
If you want to map a specific port for your container, you should map it into your local docker-compose.override.yml and not here.

File: docker-compose.yml
Comment (FabienPapet) [reply docker-compose.yml]: Forcing a port is not good (I have already something listening on this port for example), at least this should be configurable, within the override file.

Comment (M0rgan01) [reply docker-compose.yml]: On the prestashop-git service the ports are forced to 8001 and 8002, I could also have these ports used on my host. Might as well do it on the override file

**Review by FabienPapet — DISMISSED** [review]

File: docker-compose.yml
Comment (FabienPapet) [reply docker-compose.yml]: I guess this can be configurable as well, so maybe moving it to the override file can be an idea. We'll then need to update the doc or the make install create this file.

Comment (M0rgan01) [reply docker-compose.yml]: I pushed a proposal in this direction

File: Makefile
Comment (nicosomb) [inline Makefile]: Why adding this command on `make install` ? If I want to install PrestaShop locall by using `make install`, I don't want to use docker.

**Review by nicosomb — CHANGES_REQUESTED** [review]

File: Makefile
Comment (M0rgan01) [reply Makefile]: This does not start any docker container.

The docker-compose.override file will be added when installing the development environment.

This is a prerequisite for using docker later. But I can remove it, `Make docker` command is enough

Comment (nicosomb) [reply Makefile]: even if it doesn't start any container, I will have an error if docker is not installed on my computer. I think it should be removed.

File: Makefile
Comment (boherm) [inline Makefile]: `docker-up` instead?

File: Makefile
Comment (M0rgan01) [reply Makefile]: There would have been no errors, it is a command `cp docker-compose.override.yml.dist docker-compose.override.yml` which would have been launched. But I removed it

File: Makefile
Comment (M0rgan01) [reply Makefile]: It's better yes

File: Makefile
Comment (nicosomb) [reply Makefile]: you're right about the error 😉 but I think it's better to not have `docker` things during `make install`

Comment (M0rgan01) [reply Makefile]: I agree :smiley:

**Review by nicosomb — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Review by FabienPapet — DISMISSED** [review]

**Comment by matks** [comment]
@nicosomb @boherm @FabienPapet If this PR needs to be tested, you guys need to add the label (either the "Waiting for QA" for test by QA team or "Waiting for dev" combined to "Waiting for QA" for test by a developer). If no tests is needed you guys need to merge.

If you simply add the approvals but don't merge this PR will stay like this forever 😉

**Comment by FabienPapet** [comment]
Qa by dev

**Comment by SharakPL** [comment]
IMO it's not needed. We need  only a simple `.env` file for overrides + it could be used for other purposes too, not just for docker.

**Comment by tleon** [comment]
I tested this and everything seems ok. The override file is copied and the make docker-up command is working and not causing any errors. 
This PR needs to be rebased and re aprouved but otherwise it's all good for me.

**Comment by mflasquin** [comment]
Hi @M0rgan01 can you fix conflicts ? :)

**Review by boubkerbribri — CHANGES_REQUESTED** [review]
A question here, do we need to make changes in here to mention the override file ? https://github.com/PrestaShop/PrestaShop/blob/develop/.github/workflows/cron_nightly_tests_reusable.yml#L215

**Comment by matks** [comment]
Hi @M0rgan01 

Since there was no news for more than 30 days, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale. Of course if you want to continue this topic, you can reopen 😉 

Thanks!

**Pull request closed without merging**
